src/components/activity-list/ActivityList.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { SectionList, StyleSheet, View } from 'react-native';
+import sectionListGetItemLayout from 'react-native-section-list-get-item-layout';
+import ActivityIndicator from '../ActivityIndicator';
+import Spinner from '../Spinner';
+import { ButtonPressAnimation } from '../animations';
+import { CoinRowHeight } from '../coin-row/CoinRow';
+import Text from '../text/Text';
+import ActivityListEmptyState from './ActivityListEmptyState';
+import ActivityListHeader from './ActivityListHeader';
+import styled from '@/styled-thing';
+import { ThemeContextProps, useTheme } from '@/theme';
+import { useSectionListScrollToTopContext } from '@/navigation/SectionListScrollToTopContext';
+import { safeAreaInsetValues } from '@/utils';
+import { useAccountSettings, useAccountTransactions } from '@/hooks';
+import { usePendingTransactionsStore } from '@/state/pendingTransactions';
+import { TransactionSections, TransactionItemForSectionList } from '@/helpers/buildTransactionsSectionsSelector';
+
+const sx = StyleSheet.create({
+  sectionHeader: {
+    paddingVertical: 18,
+  },
+});
+
+const ActivityListHeaderHeight = 42;
+const TRANSACTION_COIN_ROW_VERTICAL_PADDING = 7;
+
+const getItemLayout = sectionListGetItemLayout({
```

Comment:
this was causing issues with the SectionList below, so I added a `@ts-expect-error` declaration for now. Maybe bumping `react-native` will fix it? But not worried about it for now.